source/server/BUILD

```diff
@@ -371,6 +371,7 @@ envoy_cc_library(
         "//source/common/stream_info:stream_info_lib",
         "//source/extensions/filters/listener:well_known_names",
         "//source/extensions/filters/network/http_connection_manager:config",
+        "//source/extensions/quic_listeners/quiche:quic_transport_socket_factory_lib",
```

Comment:
Sorry I'm confused. How does this fix the build? Doesn't this just hard code compiling H3?

Reply:
no that library is one which doesn't include quiche.
it was previously in the build and I made it selective in https://github.com/envoyproxy/envoy/pull/15540
which made the non-quic build fail because code depends on the factories existing even if they don't do anything. 
It's on my list to clean up.  IMO all code which doesn't require quiche should go into http3 to make it clear what's safe and what's not.

envoy_cc_extension(
    name = "quic_transport_socket_factory_lib",
    srcs = ["quic_transport_socket_factory.cc"],
    hdrs = ["quic_transport_socket_factory.h"],
    category = (
        # Remove QUIC as an extension
        "DELIBERATELY_OMITTED",
    ),
    security_posture = "unknown",
    tags = ["nofips"],
    undocumented = True,
    visibility = ["//visibility:public"],
    deps = [
        "//include/envoy/network:transport_socket_interface",
        "//include/envoy/server:transport_socket_config_interface",
        "//include/envoy/ssl:context_config_interface",
        "//source/common/common:assert_lib",
        "//source/extensions/transport_sockets:well_known_names",
        "//source/extensions/transport_sockets/tls:context_config_lib",
        "@envoy_api//envoy/extensions/transport_sockets/quic/v3:pkg_cc_proto",
    ],
)

Reply:
Ah OK the build was actually broken. I thought we couldn't exclude it. Yeah this is so confusing. I will help clean it up also per our DM convo.